Configure builds using yaml files

raylim opened the pull request:
fixes #87

raylim committed:
configure script reads yaml file and creates include file

create project yaml files

initProject.pl creates initial config.yaml file; configure.py sets
defaults

backward compatability ($(split_lookup.$x -> $(split.$x)))

namespaced variables

inodb commented on scripts/initProject.pl:
```diff
@@ -15,20 +15,21 @@
-CHR1_BASE_RECAL = true
+# targets_file: intervals.bed
+exome: true
+chr1_base_recal: true
```
Maybe namespace these configurations by module name so it's more clear what they are used for

inodb commented on scripts/convertProjectToYAML.py:
```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
```
I think this entire file can be removed

inodb commented on scripts/createSampleYAML.py:
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
```
This would be a good candidate for a test in Travis CI. All you need is a directory in the proper structure, so you could  make some empty files

raylim commented:
New workflow would be:

```
modules/scripts/createSampleYAML.py rawdata
modules/configure.py
```

`createSampleYAML.py` creates `sample.fastq.yaml` and the best guess for `samples.yaml`.  `configure.py` uses those files and `modules/config.yaml` to create a `config.inc` file which is read by `make`.

raylim committed:
fixed error when there are no mutations of a certain impact

raylim committed:
unset the PYTHONPATH before activating the conda env

also, reduce number of strelka cores

raylim committed:
fix #139

inodb committed:
Change clonal color in ccf plot

raylim committed:
fixed #141 and fixed #140

common calls are now filtered at the vcf level vs at the table level.

created jrflab conda env for testing

raylim committed:
added tests for configure scripts and creating sample yaml files
